pkg/apis/serving/v1alpha1/revision_types.go

```diff
@@ -273,8 +272,15 @@ func (rs *RevisionStatus) InitializeBuildCondition() {
 	revCondSet.Manage(rs).InitializeCondition(RevisionConditionBuildSucceeded)
 }
 
-func (rs *RevisionStatus) PropagateBuildStatus(bs buildv1alpha1.BuildStatus) {
-	bc := bs.GetCondition(buildv1alpha1.BuildSucceeded)
+func (rs *RevisionStatus) PropagateBuildStatus(bs duckv1alpha1.KResourceStatus) {
+	cs := bs.Conditions
+	var bc *duckv1alpha1.Condition
+	for _, c := range cs {
+		if c.Type == duckv1alpha1.ConditionSucceeded {
+			bc = &c
+			break
+		}
+	}
```

Comment:
cc @n3wscott Evidence that we need to implement `GetConditions` on our `Conditions` duck type.